### ISSUE TITLE
Add options for flipping/rotating board and flipping colors?

### ISSUE THREAD

**Issue by ikirill** [SabakiHQ/Sabaki#375]
(Feature request)

If I play a game with someone on a real board, and we both record it in an sgf file, each sgf file is "from the wrong side" for the other player. It can be a little annoying when comparing things we remembered differently, or finishing others' incomplete games from memory. I think mostly this is an issue for recording in-person club games, and I couldn't find a way to do this in Sabaki right now.

I ended up using https://homepages.cwi.nl/~aeb/go/sgfutils/html/sgfutils.html to rotate the sgf files 180 degrees, but I think it would be really nice if this was built in. Does this sound like a useful feature?

**Comment by yishn** [comment]
@fohristiwhirl I don't know what you wrote, but if you want to implement this and need help, I'd be happy to help you out

**Comment by fohristiwhirl** [comment]
Yeah I'm looking into it. Would it be enough to make a function that correctly rotated every relevant property in a node? Examples for a 90 degree clockwise rotation on a 19x19 board:

* B[bb] would become B[rb]
* AW[cc:dd] would become AW[pc:qd]

**Comment by yishn** [comment]
I think so... on top of that, we should also delete the `board` property of all nodes in the game tree which caches the board position at that node.

**Comment by fohristiwhirl** [comment]
If you like I could add a PR for a new file in the src/modules directory, perhaps.

**Comment by yishn** [comment]
> If you like I could add a PR for a new file in the src/modules directory, perhaps.

Sounds good. Thank you for the work!

**Comment by fohristiwhirl** [comment]
@ikirill - this is now merged in.

**Comment by fohristiwhirl** [comment]
@yishn - btw, I see that I left a [comment at the top](https://github.com/SabakiHQ/Sabaki/blob/master/src/modules/rotation.js#L1) which ended up being false - maybe you could remove that if you have a moment.
